shell/components/SortableTable/sorting.js

```diff
@@ -94,11 +99,28 @@ export default {
 
   methods: {
     changeSort(sort, desc) {
+      if (this.setSortFn) {
+        const realSort = this.headers.find(header => header.name === sort).sort || '';
+        const realDesc = realSort.includes(':desc') ? !desc : desc;
+        let correctedSort = realSort;
+
+        // ToDo: these if blocks are some hot garbage but they'll have to do for now...
+        if (realSort === 'creationTimestamp:desc' || realSort === 'namespace') { // ToDo: everybody hates one-off cases
+          correctedSort = `metadata.${ realSort.replace(':desc', '') }`;
+        }
+        if (isArray(realSort)) {
+          correctedSort = realSort[0].replace('$.', '').replace('[', '.').replace(']', '');
+        }
+
+        this.setSortFn(`${ realDesc ? '-' : '' }${ correctedSort }`);
+      }
       this.sortBy = sort;
       this.descending = desc;
 
       // Always go back to the first page when the sort is changed
-      this.setPage(1);
```

Comment:
Which makes sense but the passed in setSortFn should handle that in the same mutation of the resourceQuery (or whatever else it would be mutating) to avoid potential double queries/renders depending on what it's doing. Rather than rewrite the logic in the existing use-case, I simply don't use it in the new one rather than rewrite the entire component.